lib/modules/ens/index.js

```diff
@@ -216,100 +288,33 @@ class ENS {
   }
 
   configureContracts() {
-    const config = {
-      "default": {
-        "gas": "auto",
-        "contracts": {
-          "ENS": {
-            "deploy": false,
-            "silent": true
-          },
-          "ENSRegistry": {
-            "deploy": true,
-            "silent": true,
-            "args": []
-          },
-          "Resolver": {
-            "deploy": true,
-            "silent": true,
-            "args": ["$ENSRegistry"]
-          },
-          "FIFSRegistrar": {
-            "deploy": false
-          }
-        }
-      },
-      "ropsten": {
-        "contracts": {
-          "ENSRegistry": {
-            "address": "0x112234455c3a32fd11230c42e7bccd4a84e02010",
-            "silent": true
-          },
-          "Resolver": {
-            "deploy": false
-          },
-          "FIFSRegistrar": {
-            "deploy": false
-          }
-        }
-      },
-      "rinkeby": {
-        "contracts": {
-          "ENSRegistry": {
-            "address": "0xe7410170f87102DF0055eB195163A03B7F2Bff4A",
-            "silent": true
-          },
-          "Resolver": {
-            "deploy": false
-          },
-          "FIFSRegistrar": {
-            "deploy": false
-          }
-        }
-      },
-      "livenet": {
-        "contracts": {
-          "ENSRegistry": {
-            "address": "0x314159265dd8dbb310642f98f50c066173c1259b",
-            "silent": true
-          },
-          "Resolver": {
-            "deploy": false
-          },
-          "FIFSRegistrar": {
-            "deploy": false
-          }
-        }
-      }
-    };
-    config.testnet = config.ropsten;
+    this.events.request('blockchain:networkId', (networkId) => {
+      const config = Object.assign({}, DEFAULT_ENS_CONTRACTS_CONFIG, {[this.env]: ENS_CONTRACTS_CONFIG[networkId]});
```

Comment:
I think we can use a spread operator here :)

```
const config {...DEFAULT_ENS_CONTRACTS_CONFIG, ...{[this.env]: ENS_CONTRACTS_CONFIG[networkId]}
```

Reply:
I tried it and tried in the past, it doesn't work.